changelog/11970.feature.md

```diff
@@ -0,0 +1 @@
+Add a new mandatory key (`assistant_project_id`) to `config.yml` to uniquely identify assistants in deployment.
```

Comment:
That's a good point, actually the [next subtask](https://rasahq.atlassian.net/browse/ATO-681) consists of CLI adaptations (checking if the key exists and if it doesn't, emitting a `UserWarning` and generating a pseudo-random value to add to the config), do you think this PR cannot be merged unless that subtask is done as part of this PR too?

Reply:
mhm I wouldn't merge it to main (maybe fork of a longer living branch and merge to that). I don't think this change alone can be released as is, and since we want to keep main releasable that would be an issue

Reply:
Ok, I'll create a feature branch and rebase to that, if anything goes wrong in the process, I'll just resort to treating this branch as the feature branch 😄 and merge all the other subtasks to this one.